### ISSUE TITLE
Question about EPIN_ADDR

### ISSUE THREAD

**Issue by m12lrpv** [ef15c/PS2_to_USB_converter#1]
Love your project.
I've identified something that you might want to clarify...

On the default CubeMX generated code EPIN_ADDR is defined as 0x81 and passed directly to USBD_LL_Transmit
`     
#define HID_EPIN_ADDR                 0x81U
// and...
 USBD_LL_Transmit(pdev,
                       HID_EPIN_ADDR,
                       report,
                       len);`

in your project you also define HID_COMPOSITE_EPIN_BASE_ADDR as 0x81 and append 1 or 2 to it so it's sending 0x82 or 0x83 and never 0x81

Should you be decrementing HID_COMPOSITE_EPIN_BASE_ADDR  to 0x80 in the define in order to handle the later increment?
`
      USBD_LL_Transmit(pdev,
                       HID_COMPOSITE_EPIN_BASE_ADDR + hhid->interface,
                       report,
                       len);
`

**Comment by ef15c** [comment]
Hello m12rlpv,

I'm happy that you find interest in this project.

To answer your question:
In fact, the endpoints for the two interfaces are already respectively 0x81 and 0x82.

the "interface" member of the USBD_HID_COMPOSITE_HandleTypeDef structure is assigned starting from 0 in the function BuildfFSCfgDesc

As hhid->interface contains only 0 or 1, the value of HID_COMPOSITE_EPIN_BASE_ADDR (0x81) is correct.

**Comment by m12lrpv** [comment]
Thank you. Yes that makes sense. I had issues where despite only having the keyboard connected I was getting a mouse error and the keyboard wasn't working. I rewrote your code to handle just keyboard on fixed input pins and it still wouldn't work on that computer I had it connected to but would work on other computers

It turned out that I had previously plugged in a bluepill simulating a mouse with the same usb identification and the computer was recognising that the keyboard wasn't a mouse and erroring the device. I changed the USB PID and it worked. I didn't have permissions to delete the old mouse configuration in device manager.
